openhands/events/stream.py

```diff
@@ -179,13 +199,6 @@ def unsubscribe(self, subscriber_id: EventStreamSubscriber, callback_id: str):
         del self._subscribers[subscriber_id][callback_id]
 
     def add_event(self, event: Event, source: EventSource):
-        try:
-            asyncio.get_running_loop().create_task(self._async_add_event(event, source))
-        except RuntimeError:
-            # No event loop running...
-            asyncio.run(self._async_add_event(event, source))
-
-    async def _async_add_event(self, event: Event, source: EventSource):
```

Comment:
If I understand correctly, we're back to a sync `add_event`? I do like it! I'm not sure why it had switched again 😅

Reply:
async is a dark art!